lib/src/footer.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Parsing footer lines from commit messages.
+
+/// A key-value pair representing a footer line in a commit message, of the
+/// form `Key: Value`.
+#[derive(Debug, PartialEq, Clone)]
+pub struct FooterEntry(pub String, pub String);
+
+/// Parse the footer lines from a commit message; these are simple key-value
+/// pairs, separated by a colon, describing extra information in a commit
+/// message; an example is the following:
+///
+/// ```text
+/// chore: fix bug 1234
+///
+/// Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+/// tempor incididunt ut labore et dolore magna aliqua.
+///
+/// Co-authored-by: Alice <alice@example.com>
+/// Co-authored-by: Bob <bob@example.com>
+/// Reviewed-by: Charlie <charlie@example.com>
+/// Change-Id: I1234567890abcdef1234567890abcdef12345678
+/// ```
+///
+/// In this case, there are four footer lines: two `Co-authored-by` lines, one
+/// `Reviewed-by` line, and one `Change-Id` line.
+pub fn get_footer_lines(body: &str) -> Vec<FooterEntry> {
```

Comment:
Is it worth handling this case from [`git-interpret-trailers`' man page](https://git-scm.com/docs/git-interpret-trailers#:~:text=The%20%3Cvalue%3E%20may%20be%20split%20over%20multiple%20lines%20with%20each%20subsequent%20line%20starting%20with%20at%20least%20one%20whitespace%2C%20like%20the%20%22folding%22%20in%20RFC%20822.)?

> The <value> may be split over multiple lines with each subsequent line starting with at least one whitespace, like the "folding" in RFC 822. Example:
> 
> ```
> key: This is a very long value, with spaces and
>   newlines in it.
> ```

[Gerrit code](https://cs.opensource.google/gerrit/gerrit/gerrit/+/master:java/com/google/gerrit/server/ChangeUtil.java;l=171;drc=0409504cfd4fb83765950d0771cebb546c9d1d19) uses [RevCommit#getFooterLines](https://cs.opensource.google/gerrit/gerrit/gerrit/+/master:modules/jgit/org.eclipse.jgit/src/org/eclipse/jgit/revwalk/RevCommit.java;l=586;drc=b079968cdd31bb305b99f860c6b1194662619921) which seems to handle this case.

Reply:
Nice catch. Yes, I'll look into fixing this.